File: typescript/src/renderer.ts
import type { Shape } from "./shapes/shape.js"; // 使用 type-only 导入
import { newRectangle, Rectangle } from "./shapes/rectangle.js";

export class Renderer {
    private shape: Shape;

    constructor(shape: Shape) {
        this.shape = shape;
    }

    draw(): void {
        console.log("Shape drawn");
        console.log(`Its area is ${this.shape.getArea()}`);
    }
}

export function newRenderer(shape: Shape): Renderer {
    return new Renderer(shape);
}

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.Shape;

/**
 * A renderer that draws shapes and displays their area.
 */
public class Renderer {
    private final Shape shape;

    /**
     * Constructs a Renderer with the specified shape.
     *
     * @param shape the shape to be rendered
     */
    public Renderer(Shape shape) {
        this.shape = shape;
    }

    /**
     * Draws the shape and displays its area.
     */
    public void draw() {
        System.out.println("Shape drawn");
        System.out.println("Its area is " + shape.getArea());
    }
}
